src/directives/audification/audification.directive.ts

```diff
@@ -0,0 +1,48 @@
+import { ComponentRef, Directive, Host, OnDestroy, OnInit, Optional, Self } from '@angular/core';
+import { LineChartAudificationComponent } from '../../components/line-chart-audification/line-chart-audification.component';
+import { LineChartComponent } from '../../components/line-chart/line-chart.component';
+
+@Directive({
+  selector: '[appAudification]',
+})
+export class AudificationDirective implements OnInit, OnDestroy {
+  audificationComponentRef: ComponentRef<unknown> | null;
+
+  constructor(
+    // candidate host components
+    @Host() @Self() @Optional() private lineChartComponent: LineChartComponent | null,
```

Comment:
Can we have a more scale-able solution to this?

Reply:
I tried to use Union Types, but Angular gave me JIT compilation error.

I also tried to make an abstract parent class, let `LineChartComponent` and `BarChartComponent` extend it, and change the signature to `@Host() @Self() private hostComponent: ParentClass | null`, but `@Host()` injected `null` to it.

Let me know if you have other solutions that I can try.